laiaaa0/assignment4/Map.cpp

```diff
@@ -0,0 +1,106 @@
+#include "Map.h"
+
+int directions_x [] = {0,0,-1,1};
+int directions_y [] = {1,-1,0,0};
+
+Map::Map(const std::vector<std::vector<char> > & tiles) {
+  this->num_rows_ = tiles.size();
+  if (num_rows_ > 0){
+    this->num_columns_ = tiles[0].size();
+  }
+  else {
+    this->num_columns_ = 0;
+  }
+  this-> map_tiles_ = tiles;
+
+}
+
+Map::Map(const std::string & tiles_str) {
+  this->map_tiles_.clear();
+  std::vector<char> current_row;
+  for (int i = 0; i<tiles_str.size(); ++i){
+      if (tiles_str[i] == '-'){
+        this->map_tiles_.push_back(current_row);
+        current_row.clear();
+      }
+      else {
+        if (tiles_str[i]==ISLAND_T || tiles_str[i]==WATER_T)  {
+          current_row.push_back(tiles_str[i]);
+        }
+      }
+  }
+  this->num_rows_ = this->map_tiles_.size();
```

Comment:
I've seen this code to initialize `num_rows_` & `num_columns_` before :) 
Actually, it may be easier to get `map_tiles_.size()` & `map_tiles_[row].size()` when needed?